bibtex-actions-org-cite.el

```diff
@@ -0,0 +1,103 @@
+;;; bibtex-actions-org-cite.el --- Org-cite support for bibtex-actions -*- lexical-binding: t; -*-
+;;
+;; Copyright (C) 2021 Bruce D'Arcus
+;;
+;; Author: Bruce D'Arcus <https://github.com/bdarcus>
+;; Maintainer: Bruce D'Arcus <https://github.com/bdarcus>
+;; Created: July 11, 2021
+;; License: GPL-3.0-or-later
+;; Version: 0.1
+;; Homepage: https://github.com/bdarcus/bibtex-actions
+;; Package-Requires: ((emacs "26.3"))
+;;
+;; This file is not part of GNU Emacs.
+;;
+;; This program is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+;;
+;;; Commentary:
+;;
+;;  This is a small package that intergrates bibtex-actions and org-cite.  It
+;;  provides a simple org-cite processor with "follow" and "insert" capabilties.
+;;
+;;  You can configure like so:
+;;
+;;  (setq org-cite-follow-processor 'bibtex-actions-org-cite)
+;;  (setq org-cite-insert-processor 'bibtex-actions-org-cite)
+;;
+;;  It also configures Embark integration with the above.
+;;
+;;; Code:
+
+(require 'bibtex-actions)
+(require 'org)
+(require 'org-cite)
+(require 'oc-basic)
+(require 'embark)
+
+(declare-function bibtex-actions-at-point "bibtex-actions")
+(declare-function org-open-at-point "org")
+
+;TODO
+;(defvar bibtex-actions-org-cite-open-default
+
+;; Org-cite processor
+
+(defun bibtex-actions-org-cite-insert (&optional multiple)
+  "Return a list of keys when MULTIPLE, or else a key string."
+  (let ((references (bibtex-actions-read)))
+    (if multiple
+        references
+      (car references))))
+
+(defun bibtex-actions-org-cite-follow (_datum _arg)
+  "Follow processor for org-cite."
+  (call-interactively 'embark-dwim))
```

Comment:
@minad @ilupin - how does `embark-dwim` know what default function to call?

I've been driving myself crazy this morning trying to figure this out, and have fixed one piece. But still getting a void symbol error with `embark-dwim`.

Reply:
Of course the problem in the end was some stupid syntax issue with the keymap. Now fixed.